lib/msf/ui/console/command_dispatcher/db/klist.rb

```diff
@@ -118,4 +134,65 @@ def cmd_klist(*args)
   def kerberos_ticket_storage
     @kerberos_ticket_storage ||= Msf::Exploit::Remote::Kerberos::Ticket::Storage::ReadWrite.new(framework: framework)
   end
+
+  private
+
+  def ticket_status(ticket)
+    if ticket.expired?
+      '>>expired<<'
+    elsif ticket.status.blank?
+      'active'
+    else
+      ticket.status
+    end
+  end
+
+  # Sets the status of the tickets
+  #
+  # @param [Symbol] mode The status (:activate or :deactivate) to apply to the ticket(s)
+  # @param [Array<StoredTicket>] tickets The tickets which statuses are to be updated
+  # @return [Integer] Number of tickets updated
+  def set_activation_status(mode, tickets)
+    result = []
+    if mode == :activate
+      result = kerberos_ticket_storage.activate_ccache(ids: tickets.map(&:id))
+    elsif mode == :deactivate
+      result = kerberos_ticket_storage.deactivate_ccache(ids: tickets.map(&:id))
+    end
+    result.size
+  end
+
+  def ticket_search(host_ranges, id_search)
+    ticket_results = []
+
+    # Iterating over each id here since the remote db doesn't support bulk id searches
+    if id_search
+      ticket_results += id_search.flat_map do |id|
+        kerberos_ticket_storage.tickets(
+          workspace: framework.db.workspace,
+          id: id
+        )
+      end
+    else
+      each_host_range_chunk(host_ranges) do |host_search|
+        next if host_search&.empty?
+
+        ticket_results += kerberos_ticket_storage.tickets(
+          workspace: framework.db.workspace,
+          host: host_search
+        )
+      end
+      host_ranges.each { |range| range.reset unless range.nil? } # Reset the Rex::Socket::RangeWalker so it can be re-used
+    end
+
+    ticket_results.sort_by do |ticket|
+      ticket_host_address = ticket.host_address.blank? ? '0.0.0.0' : ticket.host_address
+      [::IPAddr.new(ticket_host_address).to_i, ticket.starttime.to_i]
+    end
+  rescue ActiveRecord::RecordNotFound => e
```

Comment:
This seems leaky, it should happen further down the stack - i.e. closer to the `kerberos_ticket_storage.tickets(...)` implementation.

Which should also transparently handle the remote db's exceptions that can be raised